pipeline/patch_VAE.py

```diff
@@ -167,20 +236,40 @@ def process_VAE(paths):
             3 - list of site names
 
     """
+    #TODO: add pooling datasets features and remove hardcoded normalization constants
+    channel_mean = [0.49998672, 0.007081]
+    channel_std = [0.00074311, 0.00906428]
     # these sites should be from a single condition (C5, C4, B-wells, etc..)
-    summary_folder, supp_folder, model_path, sites = paths[0], paths[1], paths[2], paths[3]
+    summary_folder, supp_folder, model_dir, sites = paths[0], paths[1], paths[2], paths[3]
+    model_path = os.path.join(model_dir, 'model.pt')
+    model_name = os.path.basename(model_dir)
+    output_dir = os.path.join(summary_folder, model_name)
+    os.makedirs(output_dir, exist_ok=True)
+
     assert len(set(site[:2] for site in sites)) == 1, \
         "Sites should be from a single well/condition"
     well = sites[0][:2]
 
     print(f"\tloading file paths {os.path.join(summary_folder, '%s_file_paths.pkl' % well)}")
     fs = pickle.load(open(os.path.join(summary_folder, '%s_file_paths.pkl' % well), 'rb'))
 
-    print(f"\tloading static patches {os.path.join(summary_folder, '%s_adjusted_static_patches.pt' % well)}")
-    dataset = torch.load(os.path.join(summary_folder, '%s_adjusted_static_patches.pt' % well))
-    dataset = rescale(dataset)
-    
-    model = VQ_VAE(alpha=0.0005, gpu=True)
+    # print(f"\tloading static patches {os.supp_dir.join(raw_dir, '%s_adjusted_static_patches.pt' % well)}")
+    # dataset = torch.load(os.supp_dir.join(raw_dir, '%s_adjusted_static_patches.pt' % well))
+    print(f"\tloading static patches {os.path.join(summary_folder, '%s_static_patches.pkl' % well)}")
+    dataset = pickle.load(open(os.path.join(summary_folder, '%s_static_patches.pkl' % well), 'rb'))
+    dataset = zscore(dataset, channel_mean=channel_mean, channel_std=channel_std)
```

Comment:
As mentioned above, I added a new function `vae_preprocess` to cover this step (and also customizable multi-channel settings).

Reply:
I guess you can handle the normalization and numpy-to-torch transition in the `vae_preprocess` function.